src/dataslice.js

```diff
@@ -81,10 +81,26 @@ export default class DataSlice {
   readUint64(offset) {
     const left = this.readUint32(offset);
     const right = this.readUint32(offset + 4);
+    let combined;
     if (this._littleEndian) {
-      return (left << 32) | right;
+      combined = left + 2 ** 32 * right;
+      if (!Number.isSafeInteger(combined)) {
+        console.error(
+          combined,
+          'exceeds MAX_SAFE_INTEGER. Precision may be lost.  Please report if you get this message to https://github.com/geotiffjs/geotiff.js/issues',
+        );
```

Comment:
I'm in favor of throwing an exception here. Uint64 is also used for file offsets, e.g where in the file the raster data is. If we are in danger territory (I know, more than 9 Petabytes are unlikely at this point) it would be better to hard fail than to continue.

Also, I'm curious: why is here `console.error` used as opposed to `console.log` below? Is the error in Int64 less severe?

Reply:
No, no reason.  I can throw an `Error`.